src/commands/redis_cmd.cc

```diff
@@ -2377,12 +2378,20 @@ class CommandSInterStore : public Commander {
 class CommandZAdd : public Commander {
  public:
   Status Parse(const std::vector<std::string> &args) override {
-    if (args.size() % 2 != 0) {
-      return Status(Status::RedisParseErr, errInvalidSyntax);
+    unsigned index = 2;
+    parseFlags(args, index);
+    if (auto s = validateFlags(); !s.IsOK()) {
+      return s;
+    }
+    if (auto left = (args.size() - index); left > 0) {
```

Comment:
How about left = 0? if requests have only flags but no member and score?

Reply:
It's a bug. It's fixed now.